src/cljs/org/broadinstitute/firecloud_ui/page/workspace/data/tab.cljs

```diff
@@ -176,21 +173,19 @@
                             [DataImporter {:dismiss #(swap! state dissoc :show-import?)
                                            :workspace-id workspace-id
                                            :this-realm this-realm
-                                           :reload-data-tab (fn [entity-type]
-                                                              (swap! state dissoc :entity-list :entity-types)
-                                                              (react/call :refresh this entity-type))}])}])
+                                           :reload-data-tab #(react/call :refresh this %)}])}])
         (when (:show-delete? @state)
           [DataDeleter {:dismiss-self #(swap! state dissoc :show-delete?)
-                        :entity-list entity-list
+                        :get-entity-list #(react/call :get-entity-list (@refs "entity-table"))
                         :workspace-id workspace-id
                         :reload #(react/call :refresh this)}])
         (cond
           server-error
           (style/create-server-error-message server-error)
-          (some nil? [locked? entity-list])
-          [:div {:style {:textAlign "center"}} [comps/Spinner {:text "Loading entities..."}]]
+          (nil? locked?)
+          [:div {:style {:textAlign "center"}} [comps/Spinner {:text "Fetching entity types..."}]]
```

Comment:
This really isn't honest anymore—we're really just checking if the workspace is locked.